File: JavaScript/PM_68644_두 개 뽑아서 더하기.js
function solution(numbers) {
  const temp = [];

  for (let i = 0; i < numbers.length; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      temp.push(numbers[i] + numbers[j]);
    }
  }
  // 전개구문을 통하여 값들만 answer에 넣어준다.
  // Set은 중복제거 역할
  const answer = [...new Set(temp)];

  // return 값이 양수면 자리 바꿈 (오름차순)
  console.log(answer.sort((a, b) => a - b));
  return answer.sort((a, b) => a - b);
}

const numbers = [2, 1, 3, 4, 1];
solution(numbers);

File: Java/PM_68644_두 개 뽑아서 더하기.java
package problemSolve;

import java.util.ArrayList;
import java.util.Collections;

public class PM_68644 {
	//PM(프로그래머스) / 68644번 / 두 개 뽑아서 더하기 
	public static void main(String[] args) {
		
		int [] numbers = {5,0,2,7};
		PM_68644 c = new PM_68644();
		
		c.solution(numbers);
	}
	
    public int[] solution(int[] numbers) {
        
        ArrayList<Integer> al = new ArrayList<>();
        
        for(int i=0; i<numbers.length; i++) {
        	for(int j= i+1; j<numbers.length; j++) {
        		int sum = numbers[i] + numbers[j]; // 반복문을 돌면서 더해준다. 
        		
        		// sum이 al에 포함되어 있지 않으면 추가. 중복된 값은 넣지 않는다. 
        		if(al.contains(sum)) {
        			continue;
        		}
        		else {
        			al.add(sum);        			
        		}
        	}
        }        
        int[] answer = new int[al.size()];
        Collections.sort(al); // al을 오름차순 정렬한다.
        
        for(int i=0; i<al.size(); i++) {
        	answer[i] = al.get(i);
        	System.out.println(answer[i]);
        }
        return answer;
    }
}
